### ISSUE TITLE
attempting to read past EOF throws error.ServerMalformedResponse

### ISSUE THREAD

**Issue by nektro** [alexnask/iguanaTLS#7]
it should instead return the `usize` value of `0` instead of an error.

**Comment by alexnask** [comment]
@nektro 
Can you post some example code? 
Ive tried this both with a TCP connection, which just hangs as expected and the benchmark's replaying tool, by returning zero from the reader after the handshake is done, and I am getting zero returned as expected.

**Comment by nektro** [comment]
no worries, here's a proof of concept https://github.com/nektro/iguanatls-bug-7. reproduce with:

```
zigmod fetch
zig build run
```

on my machine,
```
❯ zig version
0.8.0-dev.1071+fdc875ed0
```

```
❯ uname -a
Linux 4.19.0-13-amd64 #1 SMP Debian 4.19.160-2 (2020-11-28) x86_64 GNU/Linux
```

**Comment by alexnask** [comment]
Fixed in 58f72f64000791bbaef535acab9db726b88f7cb4
